lib/active_merchant/billing/gateways/commerce_hub.rb

```diff
@@ -223,9 +223,22 @@ def add_payment_token(source, payment, options)
         end
       end
 
+      def add_decypted_wallet(source, payment, options)
+        source[:sourceType] = 'DecryptedWallet'
+        source[:card] = {
+          cardData: payment.number,
+          expirationMonth: (format(payment.month, :two_digits) if payment.month),
```

Comment:
❓ Question:

Any reason to think that a payment method could have `number` but not `month` or `year` ?, given that you are not raising an error here you end-up sending an incomplete `:card` object to the gateway